src/cache/first-facets-distribution.ts

```diff
@@ -1,14 +1,20 @@
-import { FacetDistribution } from '../types'
+import { FacetDistribution, SearchContext } from '../types'
+import { MeiliParamsCreator } from '../adapter'
 
-export function cacheFirstFacetDistribution(
-  defaultFacetDistribution: FacetDistribution,
-  searchResponse: any
-): FacetDistribution {
-  if (
-    searchResponse.query === '' &&
-    Object.keys(defaultFacetDistribution).length === 0
-  ) {
-    return searchResponse.facetDistribution
-  }
-  return defaultFacetDistribution
+export async function cacheFirstFacetDistribution(
+  searchResolver: any,
+  searchContext: SearchContext
+): Promise<FacetDistribution> {
+  const meilisearchParams = MeiliParamsCreator(searchContext)
+  meilisearchParams.addFacets()
+  meilisearchParams.addAttributesToRetrieve()
```

Comment:
Which attributesToRetrieve this query will receive? All of it? If yes, we could try to receive no attributes at all, since we just want the facets... This will reduce the amount of data handled by the client side.